src/main/java/com/github/princesslana/eriscasper/repository/GuildRepository.java

```diff
@@ -0,0 +1,16 @@
+package com.github.princesslana.eriscasper.repository;
+
+import com.github.princesslana.eriscasper.data.Snowflake;
+import com.github.princesslana.eriscasper.data.resource.Channel;
+import com.github.princesslana.eriscasper.data.resource.Guild;
+import io.reactivex.Maybe;
+
+public interface GuildRepository {
+  Maybe<Guild> getGuild(Snowflake id);
+
+  Maybe<Channel> getChannel(Snowflake id);
+
+  Maybe<Guild> getGuildFromChannel(Snowflake channelId);
```

Comment:
Is the method better at this level, or would be it better on a 'Channels' or 'Guilds' class? (like https://github.com/princesslana/ErisCasper.java/blob/master/src/main/java/com/github/princesslana/eriscasper/data/Users.java)

Should expect the user, if they have a channel id, to fetch the `Channel`, then get the guild via `Channel#getGuildId`?